### ISSUE TITLE
deps: fix deps issues for CI testing

### ISSUE THREAD

**Issue by anguillanneuf** [googleapis/java-pubsublite#73]
Error in tests:

```
[INFO] --- maven-dependency-plugin:3.1.2:analyze (default-cli) @ google-cloud-pubsublite ---
[WARNING] Used undeclared dependencies found:
[WARNING]    com.google.code.findbugs:jsr305:jar:3.0.2:compile
[WARNING]    com.google.api:api-common:jar:1.9.0:compile
[WARNING]    com.google.guava:guava:jar:29.0-android:compile
[WARNING]    junit:junit:jar:4.13:test
[WARNING]    org.threeten:threetenbp:jar:1.4.4:compile
[WARNING]    io.grpc:grpc-api:jar:1.29.0:compile
[WARNING]    com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[WARNING]    com.google.protobuf:protobuf-java:jar:3.12.0:compile
[WARNING]    io.grpc:grpc-auth:jar:1.29.0:compile
[WARNING]    io.grpc:grpc-core:jar:1.29.0:compile
[WARNING]    org.hamcrest:hamcrest-core:jar:1.3:test
[WARNING]    com.google.auto.value:auto-value-annotations:jar:1.7.2:compile
[WARNING]    com.google.auth:google-auth-library-oauth2-http:jar:0.20.0:compile
[WARNING]    com.google.api.grpc:proto-google-cloud-pubsub-v1:jar:1.87.0:compile
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.flogger:flogger-system-backend:jar:0.5.1:test
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:analyze (default-cli) on project google-cloud-pubsublite: Dependency problems found -> [Help 1]
```

Steps to reproduce: `mvn dependency:analyze`. 

Question to @stephaniewang526: Should these warnings cause deps test failure ([1](https://source.cloud.google.com/results/invocations/6b70ba42-f4d3-4bd3-90ff-f3f1af1e0a2b/targets/cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-pubsublite%2Fpresubmit%2Fdependencies/log))? 

Additionally, it would be nice that users don't need to add `flogger` as a runtime dependency. 

https://github.com/googleapis/java-pubsublite/blob/209cae517a0b1d90b75c9d978cc36e0b6b4d8fa9/samples/snippets/pom.xml#L56-L62

**Comment by stephaniewang526** [comment]
Yes they typically do... these deps need to be declared in the pubsub-lite client pom.

**Comment by anguillanneuf** [comment]
Fixed by #85
